All_Relationships/3.model-label-functions.py

```diff
@@ -149,59 +148,105 @@
 
 
 # In[14]:
 
 
-get_ipython().run_cell_magic('time', '', '# Model each label function and the underlying correlation structure\ngen_model = GenerativeModel(lf_propensity=True)\ngen_model.train(\n    L_train,\n    epochs=10,\n    decay=0.95,\n    step_size=0.1 / L_train.shape[0],\n    reg_param=1e-6,\n    threads=50,\n    deps=deps\n)')
+indep_models = []
+for columns in lfs_columns:
+    #Conditionally independent Generative Model
+    indep_gen_model = GenerativeModel()
+    indep_gen_model.train(
+        L_train[:, columns],
+        epochs=10,
+        decay=0.95,
+        step_size=0.1 / L_train[:, columns].shape[0],
+        reg_param=1e-6,
+        threads=50,
+    )
+    indep_models.append(indep_gen_model)
 
 
-# # Generative Model Statistics
+# In[15]:
 
-# Now that both models have been trained, the next step is to generate some statistics about each model. The two histograms below show a difference between both models' output. The conditionally independent model (CI) predicts more negative candidates compared to the dependancy aware model (DA).
 
-# In[15]:
+dep_models = []
+for columns in lfs_columns:
+    # select the dependancies from the label matrix
+    ds = DependencySelector()
+    deps = ds.select(L_train[:, columns], threshold=0.1)
+    print(len(deps))
+
+    # Model each label function and the underlying correlation structure
+    gen_model = GenerativeModel(lf_propensity=True)
+    gen_model.train(
+        L_train[:, columns],
+        epochs=10,
+        decay=0.95,
+        step_size=0.1 / L_train[:, columns].shape[0],
+        reg_param=1e-6,
+        threads=50,
+        deps=deps
+    )
+    
+    dep_models.append(gen_model)
 
 
-# Generate Statistics of Generative Model
-indep_learned_stats_df = indep_gen_model.learned_lf_stats()
-learned_stats_df = gen_model.learned_lf_stats()
+# # Generative Model Statistics
 
+# Now that both models have been trained, the next step is to generate some statistics about each model. The two histograms below show a difference between both models' output. The conditionally independent model (CI) predicts more negative candidates compared to the dependancy aware model (DA).
 
 # In[16]:
 
 
-get_ipython().run_cell_magic('time', '', 'train_marginals_indep = indep_gen_model.marginals(L_train)\ntrain_marginals = gen_model.marginals(L_train)')
+get_ipython().run_cell_magic('time', '', 'train_marginals_indep_df = pd.DataFrame(\n    np.array([model.marginals(L_train[:, columns]) for model, columns in zip(indep_models, lfs_columns)]).T,\n    columns=model_names\n)\n\ntrain_marginals_dep_df = pd.DataFrame(\n    np.array([model.marginals(L_train[:, columns]) for model, columns in zip(dep_models, lfs_columns)]).T,\n    columns=model_names\n)')
 
 
 # In[17]:
 
 
-plt.hist(train_marginals_indep, bins=20)
-plt.title("CI Training Set Marginals")
-plt.ylabel("Frequency")
-plt.xlabel("Probability of Positive Class")
+f, axs = plt.subplots(len(lfs_columns), 1, figsize=(10, 5), sharex=True)
```

Comment:
I might not understand the meaning of the plots here when we went over them together so I don't know if you want to add comments here/change the title of the axes